mucommander-core/src/main/java/com/mucommander/ui/main/MainFrame.java

```diff
@@ -247,20 +265,22 @@ public Insets getInsets() {
         YBoxPanel southPanel = new YBoxPanel();
         southPanel.addSpace(2);
 
-        // Add status bar
-        this.statusBar = new StatusBar(this);
-        southPanel.add(statusBar);
-
-        // Show command bar only if it hasn't been disabled in the preferences
-        this.commandBar = new CommandBar(this);
-        // Note: CommandBar.setVisible() has to be called no matter if CommandBar is visible or not, in order for it to be properly initialized
-        this.commandBar.setVisible(MuConfigurations.getPreferences().getVariable(MuPreference.COMMAND_BAR_VISIBLE, MuPreferences.DEFAULT_COMMAND_BAR_VISIBLE));
-        southPanel.add(commandBar);
-        insetsPane.add(southPanel, BorderLayout.SOUTH);
+        executor.execute(() -> {
```

Comment:
I wonder if we really save much by using several threads here and above - did you check how much we get from that?

Reply:
Don't remember the numbers, it was enough to grab my attention :) command bar and status bar have their icons etc, so it adds a delay if we do it in series. On my relatively slow macbook, sometimes I see that main window is presented and status bar is being added a moment later. So having parallel work on it gives that 'responsiveness' feeling.

Reply:
I see but I wonder whether it's a better approach - I understand the need to reduce the time until users can a fully function window but getting "half baked" window more quickly is not necessarily better I think

Reply:
usually the window is fully loaded quicker, sometimes for blink of second you see those menus appear with slight of latency (And you have to remember that I'm using a macbook from 2012 :) ).